### PR TITLE
Create user model and relation one to many with book

### PR DESCRIPTION
**Pull request by JJhoan**
## Summary
-Added user entity
-One to many  relationship was created
-Added methods to add and delete books in user entity

## Trello Card
https://trello.com/c/8lZeztLA/19-crear-modelo-de-user

### CODE REVIEW

File: src/main/java/com/wolox/training/model/User.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+
+@Entity(name = "Usser")
```
Comment (IgnOliveto) [inline src/main/java/com/wolox/training/model/User.java:21]: I think the name of the table is wrong.

File: src/main/java/com/wolox/training/model/User.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Long idUser;
+    @NotNull
+    private String username;
+    @NotNull
+    private String name;
+    @NotNull
+    private LocalDate birthdate;
+    @OneToMany
+    @JoinColumn(name= "user_id")
```
Comment (IgnOliveto) [inline src/main/java/com/wolox/training/model/User.java]: You would have to define the type of the relationship and type of cascade

File: src/main/java/com/wolox/training/model/Book.java [hunk 3]
```diff
@@ -29,12 +32,15 @@
     private Integer pages;
     @NotNull
     private String isbn;
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    private User user;
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/model/Book.java]: you don't need the `@JoinColum` annotation

File: src/main/java/com/wolox/training/model/User.java (hunk 2 above)
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/model/User.java]: you don't need de `@JoinColum` annotation for this

**Review by jeymarsonwolox — APPROVED** [review]

File: src/main/java/com/wolox/training/model/Book.java [hunk 4]
```diff
@@ -29,12 +32,15 @@
     private Integer pages;
     @NotNull
     private String isbn;
+    @ManyToOne
+    @JoinColumn(name = "user_id")
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/model/Book.java]: use `@ManyToMany`, one user can be associated with many books and one book can be associated with many users

File: src/main/java/com/wolox/training/model/User.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Long idUser;
+    @NotNull
+    private String username;
+    @NotNull
+    private String name;
+    @NotNull
+    private LocalDate birthdate;
+    @OneToMany
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/model/User.java]: use `@ManyToMany`, one user can be associated with many books and one book can be associated with many users

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]

File: src/main/java/com/wolox/training/model/Book.java [hunk 6]
```diff
@@ -29,12 +36,14 @@
     private Integer pages;
     @NotNull
     private String isbn;
+    @ManyToMany
+    private List<User> users;
```
Comment (jeymarsonwolox) [inline src/main/java/com/wolox/training/model/Book.java:41]: add `@JsonIgnore` to avoid a cyclic reference when querying

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]

File: src/main/java/com/wolox/training/model/Book.java (hunk 6 above)
Comment (jeymarsonwolox) [reply src/main/java/com/wolox/training/model/Book.java:41]: @JJhoan  make this please

**Review by jeymarsonwolox — DISMISSED** [review]

**Review by IgnOliveto — DISMISSED** [review]

**Review by jeymarsonwolox — APPROVED** [review]

**Review by IgnOliveto — DISMISSED** [review]

**Review by IgnOliveto — APPROVED** [review]

**Pull request merged**